CMakeLists.txt

```diff
@@ -1,81 +1,472 @@
+#* for more details.
+#*
+#* You should have received a copy of the GNU Lesser General Public
+#* License along with FMS.  If not, see <http://www.gnu.org/licenses/>.
+#***********************************************************************
+
+# Copyright (c) GFDL, @underwoo
+
+cmake_minimum_required(VERSION 3.12 FATAL_ERROR)
+
+# Define the CMake project
+project(FMS
+  VERSION 2020.03.001
+  # Forked in 2020 at: https://github.com/ACCESS-NRI/FMS/commit/e8940fe90d68c3dc7c0d6bf1b8f552a577251754
+  DESCRIPTION "ACCESS-NRI fork of GFDL FMS Library"
+  HOMEPAGE_URL "https://github.com/ACCESS-NRI/FMS/"
+  LANGUAGES C Fortran)
+
+include(GNUInstallDirs)
+
+if(NOT CMAKE_BUILD_TYPE MATCHES "^(Debug|Release|RelWithDebInfo|MinSizeRel)$")
+  message(STATUS "Setting build type to 'Release' as none was specified.")
+  set(CMAKE_BUILD_TYPE
+      "Release"
+      CACHE STRING "Choose the type of build." FORCE)
+  set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
+endif()
+
+if(NOT CMAKE_C_COMPILER_ID MATCHES "^(Intel|GNU|Clang|IntelLLVM)$")
+  message(
+    WARNING "Compiler not officially supported: ${CMAKE_C_COMPILER_ID}")
+endif()
+
+if(NOT CMAKE_Fortran_COMPILER_ID MATCHES "^(Intel|GNU|IntelLLVM)$")
+  message(
+    WARNING "Compiler not officially supported: ${CMAKE_Fortran_COMPILER_ID}")
+endif()
+
+# Append directory that contains CMake Modules for building FMS
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
+
+################################################################################
+# Fortran
+################################################################################
+
+if(CMAKE_Fortran_COMPILER_ID MATCHES "GNU")
+
+  # Precision-based Fortran compiler flags
+  set(r8_flags "-fdefault-real-8 -fdefault-double-8") # Fortran flags for 64BIT precision
+
+  # Copied from MOM5/bin/mkmf.template.nci.gfortran
+  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fcray-pointer -fdefault-real-8 -ffree-line-length-none -fno-range-check -Waliasing -Wampersand -Warray-bounds -Wcharacter-truncation -Wconversion -Wline-truncation -Wintrinsics-std -Wsurprising -Wno-tabs -Wunderflow -Wunused-parameter -Wintrinsic-shadow -Wno-align-commons")
+  set(CMAKE_Fortran_FLAGS_RELEASE "-O2")
+  set(CMAKE_Fortran_FLAGS_DEBUG "-O0 -g -W -fbounds-check")
+
+elseif(CMAKE_Fortran_COMPILER_ID STREQUAL "Intel")
+
+  # Precision-based Fortran compiler flags
+  set(r4_flags "-real-size 32") # Fortran flags for 32BIT precision
+  set(r8_flags "-real-size 64") # Fortran flags for 64BIT precision
+  set(r8_flags "${r8_flags} -no-prec-div -no-prec-sqrt")
+
+  # Copied from MOM5/bin/mkmf.template.nci
+  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fno-alias -safe-cray-ptr -fpe0 -ftz -assume byterecl -i4 -r8 -traceback -nowarn -check noarg_temp_created -assume nobuffered_io -convert big_endian -grecord-gcc-switches -align all")
+  set(CMAKE_Fortran_FLAGS_RELEASE "-g3 -O2 -xCORE-AVX2 -debug all -check none")
```

Comment:
Intel compilers are etting `-g3` in `RELEASE` so isn't consistent with above. 

I don't mind if this the default across all the compilers but it is still better to use `RelWithDebInfo` for this I think.

Reply:
Are you comparing GNU versus Intel compiler flags? Are they expected to be consistent? In this PR, let's keep the compiler flags identical to MOM5. Diverging from MOM5 compiler flags should be a separate issue.